src/interface/eager.jl

```diff
@@ -51,6 +51,26 @@ function Base.reduce(op::Function, bc::Broadcasted{FinchStyle{N}}; dims=:, init
     end
 end
 
+
+function tensordot(A::Tensor, B::Tensor, idxs; kw...)
+    bc_A = broadcasted(identity, A)
+    bc_B = broadcasted(identity, B)
+    tensordot(bc_A, bc_B, idxs; kw...)
+end
+
+function tensordot(bc_A::Broadcasted{FinchStyle{N1}}, bc_B::Broadcasted{FinchStyle{N2}}, idxs; mult_op=*, add_op=+, init = initial_value(mult_op, Float64)) where {N1, N2}
```

Comment:
Because users can now use a lazy fusion interface, maybe we don't need to support e.g. reduce(broadcast) or tensordot(broadcast), especially if it's only going to fall back to the lazy fusion interface.

Reply:
Makes sense to me, that feels cleaner. I'll remove this version.